xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/BaseIT.java

```diff
@@ -0,0 +1,16 @@
+package org.xrpl.xrpl4j.tests;
+
+import org.xrpl.xrpl4j.client.XrplClient;
+import org.xrpl.xrpl4j.tests.environment.XrplEnvironment;
+
+public class BaseIT extends AbstractIT {
```

Comment:
From what I can tell, this class is not necessary. We can move the internals of this class up into `AbstractIT` and then remove this class entirely. Once done, the override in `NfTokenIT` will still work properly. Here's what the values in `AbstractIT` should look like:

```java
private static XrplEnvironment xrplEnvironment = XrplEnvironment.getConfiguredEnvironment();

protected XrplClient xrplClient = xrplClient();

protected XrplEnvironment xrplEnvironment() {
  return xrplEnvironment;
}

protected XrplClient xrplClient() {
    return xrplEnvironment().getXrplClient();
}
```

You'll then want to do the following:

1. Remove any references to `xrpClient()` and replace with `xrpClient`.
2. Get rid of the client declaration in `NfTokenIT` near line 39. The value in `AbstractIT` will work fine.

Reply:
@sappenin I actually suggested this pattern to mukul, but I could be wrong. I wanted to avoid calling `XrplEnvironment.getConfiguredEnvironment` statically on all tests if you're just running a test that uses a custom environment. For example, if you only wanted to run the NFT tests and we call `XrplEnvironment.getConfiguredEnvironment()` in `AbstractIT`, and the configured environment is local rippled, your tests will end up spinning up a rippled container even though you overrode the `XrplEnvironment` in your test class.

If you only call `XrplEnvironment.getConfiguredEnvironment()` in `BaseIT`, a rippled container only gets started up if you are running a test that extends `BaseIT`, no?

Reply:
Yes, I agree with your goal @nkramer44 -- see my latest comment and latest commit. I think what I pushed accomplishes both of our goals.